packages/react/src/components/Combobox/Option/Option.tsx

```diff
@@ -0,0 +1,129 @@
+import React, {
+  forwardRef,
+  useContext,
+  useEffect,
+  useId,
+  useMemo,
+} from 'react';
+import cn from 'classnames';
+import { useMergeRefs } from '@floating-ui/react';
+
+import { ComboboxContext } from '../Combobox';
+import { Label } from '../../Typography';
+import ComboboxCheckbox from '../internal/ComboboxCheckbox';
+import { omit } from '../../../utilities';
+
+import classes from './Option.module.css';
+import ComboboxOptionDescription from './Description/Description';
+
+export type ComboboxOptionProps = {
+  /**
+   * The value returned when the item is selected
+   */
+  value: string;
+  /**
+   * The index of the item in the list, will be overwritten by Combobox.
+   */
+  description?: string;
+  /**
+   * The text displayed in the input or in the chips when the item is selected.
+   * Required if children is not composed of strings only.
+   */
+  displayValue?: string;
+} & React.ButtonHTMLAttributes<HTMLButtonElement>;
+
+export const ComboboxOption = forwardRef<
+  HTMLButtonElement,
+  ComboboxOptionProps
+>(({ value, description, children, ...rest }, ref) => {
+  const labelId = useId();
+  const generatedId = useId();
+
+  const context = useContext(ComboboxContext);
+  if (!context) {
+    throw new Error('ComboboxOption must be used within a Combobox');
+  }
+  const {
+    selectedOptions,
+    activeIndex,
+    setActiveItem,
+    onOptionClick,
+    multiple,
+    size,
+    listRef,
+    optionValues,
+  } = context;
+
+  const index = useMemo(
+    () => optionValues.indexOf(value),
+    [optionValues, value],
+  );
+
+  const combinedRef = useMergeRefs([
+    (node: HTMLElement | null) => {
+      listRef.current[index] = node;
+    },
+    ref,
+  ]);
+
+  if (typeof index !== 'number') {
```

Comment:
`indexOf` returns `-1` if it didn't find anything so error would never be thrown.
```suggestion
  if (index === -1) {
```